core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -425,10 +425,9 @@ private[spark] class TaskSchedulerImpl(
           }
         }
       } else {
-        // No task sets are active but we still got an error. Just exit since this
-        // must mean the error is during registration.
-        // It might be good to do something smarter here in the future.
-        throw new SparkException(s"Exiting due to error from cluster scheduler: $message")
+        // No task sets are active but we still got an error. Just log the error
+        // since this must mean the error is during registration.
+        logError(s"Exiting due to error from cluster scheduler: $message")
```

Comment:
Do you mean add a catch in the dead method of the SparkDeploySchedulerBackend? And then...log the error within the catch block? Is not the same that just log the error in the TaskSchedulerImpl?